File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/domain/usecase/DeleteHelpcardsByUnlockStateUseCase.java
package space.lopatkin.spb.helpboardgamecard.domain.usecase;

import space.lopatkin.spb.helpboardgamecard.domain.repository.AppRepository;

public class DeleteHelpcardsByUnlockStateUseCase {

    private AppRepository repository;

    public DeleteHelpcardsByUnlockStateUseCase(AppRepository repository) {
        this.repository = repository;
    }

    public void execute() {
        repository.deleteAllUnlockHelpcards();
    }

}

File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/domain/usecase/DeleteHelpcardsByUnlockStateUseCase.kt
package space.lopatkin.spb.helpboardgamecard.domain.usecase

import space.lopatkin.spb.helpboardgamecard.domain.repository.AppRepository

class DeleteHelpcardsByUnlockStateUseCase(private val repository: AppRepository) {

    fun execute() {
        repository.deleteAllUnlockHelpcards()
    }

}
